client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchRoundDataEntry.test.tsx

```diff
@@ -0,0 +1,458 @@
+import React from 'react'
+import { render, screen, within, waitFor } from '@testing-library/react'
+import { QueryClientProvider } from 'react-query'
+import userEvent from '@testing-library/user-event'
+import { ToastContainer } from 'react-toastify'
+import { Classes } from '@blueprintjs/core'
+import BatchRoundDataEntry from './BatchRoundDataEntry'
+import { batchesMocks } from '../_mocks'
+import { IBatches, IBatchResultTallySheet } from '../useBatchResults'
+import { IContest } from '../../../types'
+import {
+  withMockFetch,
+  findAndCloseToast,
+  serverError,
+  createQueryClient,
+} from '../../testUtilities'
+import { contestMocks } from '../../AuditAdmin/useSetupMenuItems/_mocks'
+
+const apiCalls = {
+  getJAContests: (response: { contests: IContest[] }) => ({
+    url: `/api/election/1/jurisdiction/1/contest`,
+    response,
+  }),
+  getBatches: (response: IBatches) => ({
+    url: '/api/election/1/jurisdiction/1/round/round-1/batches',
+    response,
+  }),
+  putBatchResults: (
+    batchId: string,
+    resultTallySheets: IBatchResultTallySheet[]
+  ) => ({
+    url: `/api/election/1/jurisdiction/1/round/round-1/batches/${batchId}/results`,
+    options: {
+      method: 'PUT',
+      body: JSON.stringify(resultTallySheets),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    },
+    response: { status: 'ok' },
+  }),
+  finalizeBatchResults: () => ({
+    url: `/api/election/1/jurisdiction/1/round/round-1/batches/finalize`,
+    options: { method: 'POST' },
+    response: { status: 'ok' },
+  }),
+}
+
+const batchesWithResults = (resultTallySheets: IBatchResultTallySheet[]) => [
+  {
+    ...batchesMocks.emptyInitial.batches[0],
+    resultTallySheets,
+  },
+  ...batchesMocks.emptyInitial.batches.slice(1),
+]
+
+const renderComponent = () =>
+  render(
+    <QueryClientProvider client={createQueryClient()}>
+      <BatchRoundDataEntry
+        electionId="1"
+        jurisdictionId="1"
+        roundId="round-1"
+      />
+      <ToastContainer />
+    </QueryClientProvider>
+  )
+
+describe('Batch comparison data entry', () => {
```

Comment:
I've essentially adapted the existing tests to work for the new UI. But there are lots of new interactions that I want to test more deeply. Took note of this in https://github.com/votingworks/arlo/issues/1655